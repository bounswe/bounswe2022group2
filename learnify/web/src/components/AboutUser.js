import React, { useState } from 'react';
import './component_styles.css';

function AboutUser(props) {
  const [aboutText, setAboutText] = useState(props.bio);
  const [isEditing, setIsEditing] = useState(false);
  const [username, setUsername] = useState(localStorage.getItem("username"));
  
  function handleEditButtonClick() {
    setIsEditing(true);
  }

  async function handleSaveButtonClick() {
    setIsEditing(false);
    try {
      const response = await fetch(`${process.env.REACT_APP_BACKEND_BASE_URL}user/`, {
        method: "PUT",
        headers: {
          "Content-Type": "application/json",
          "Authorization": `${localStorage.getItem("token")} `,
        },
        body: JSON.stringify({
          bio: aboutText,
        }),
      });
      if (!response.ok) {
        throw new Error("Failed to update bio");
      }
      window.location.reload();
    } catch (error) {
      console.error(error);
    }
  }

  function EditButton({ onClick }) {
    if(props.userviewed === username){
    return (
      <button className="edit-button" onClick={onClick}>
        Edit
      </button>
    );}
  }

  return (
    <div className="about-user-container">
      <div className="about-user-frame">
        {isEditing ? (
          <>
            <textarea
              className="about-text-input"
              value={aboutText}
              onChange={(event) => setAboutText(event.target.value)}
            />
            <div className="space-3" />
            <button className="save-button" onClick={handleSaveButtonClick}>
              Save
            </button>
          </>
        ) : (
          <>
            <label className="navBarText2">Biography</label>
            <div className="space-3" />
            <label className="about-text">{props.bio}</label>
            <div className="space-3" />
            <EditButton onClick={handleEditButtonClick} />
          </>
        )}
      </div>
    </div>
  );
}

export default AboutUser;
