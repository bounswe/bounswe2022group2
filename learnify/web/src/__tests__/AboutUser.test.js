import React from "react";
import AboutUser from "../components/AboutUser";
import { screen, render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe("AboutUser", () => {
    test('should render page layout correctly', async () => {
        const component = renderer.create(<BrowserRouter><AboutUser /></BrowserRouter>).toJSON();
        expect(component).toMatchSnapshot();
    });
});

