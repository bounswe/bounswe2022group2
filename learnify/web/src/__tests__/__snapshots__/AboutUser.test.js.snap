// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUser should render page layout correctly 1`] = `
<div
  className="about-user-container"
>
  <div
    className="about-user-frame"
  >
    <label
      className="navBarText2"
    >
      Biography
    </label>
    <div
      className="space-3"
    />
    <label
      className="about-text"
    />
    <div
      className="space-3"
    />
  </div>
</div>
`;
